File: recyclermvp/src/main/java/com/kirchhoff/recyclermvp/RepositoriesRecyclerAdapter.java
package com.kirchhoff.recyclermvp;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;


/**
 * @author Kirchhoff-
 */

public class RepositoriesRecyclerAdapter extends RecyclerView.Adapter<RepositoryViewHolder> {

    private final RepositoriesListPresenter presenter;


    public RepositoriesRecyclerAdapter(RepositoriesListPresenter repositoriesListPresenter) {
        this.presenter = repositoriesListPresenter;
    }

    @Override
    public RepositoryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new RepositoryViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.r_mvp_repository, parent, false));
    }

    @Override
    public void onBindViewHolder(RepositoryViewHolder holder, int position) {
        presenter.onBindRepositoryRowViewAtPosition(position, holder);
    }

    @Override
    public int getItemCount() {
        return presenter.getReposotoriesRowsCount();
    }
}

File: recyclermvp/src/main/java/com/kirchhoff/recyclermvp/RepositoriesRecyclerAdapter.kt
package com.kirchhoff.recyclermvp

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

class RepositoriesRecyclerAdapter(private val presenter: RepositoriesListPresenter) : RecyclerView.Adapter<RepositoryViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            RepositoryViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.r_mvp_repository, parent, false))

    override fun onBindViewHolder(holder: RepositoryViewHolder, position: Int) {
        presenter.onBindRepositoryRowViewAtPosition(position, holder)
    }

    override fun getItemCount() = presenter.getReposotoriesRowsCount()
}
